src/quantem/imaging/lattice.py

```diff
@@ -0,0 +1,2961 @@
+import numpy as np
+import torch
+from numpy.typing import NDArray
+from scipy.optimize import least_squares
+
+from quantem.core.datastructures.dataset2d import Dataset2d
+from quantem.core.datastructures.vector import Vector
+from quantem.core.io.serialize import AutoSerialize
+from quantem.core.visualization import show_2d
+
+
+class Lattice(AutoSerialize):
```

Comment:
File i/o seems to work properly:
<img width="584" height="913" alt="image" src="https://github.com/user-attachments/assets/1814820e-2fb4-4750-996c-d8c0f3171f29" />
Did you write a test?  What about when using the torch parts, do they work with autoserialize?